CMakeExt/LAPACK.cmake

```diff
@@ -4,7 +4,7 @@ find_package(BLAS)
 find_package(LAPACK)
 
 if (BLAS_FOUND)
-  check_library_exists(${BLAS_LIBRARIES} "cblas_sgemm" BLAS_IS_CBLAS)
+  check_library_exists(${BLAS_LIBRARIES} "cblas_sgemm" "C" BLAS_IS_CBLAS)
```

Comment:
While we're at it: why don't we restrict the search for LAPACK to when support for LAPACK has been requested?